YHP_website/src/main/java/storage/UserCache.java

```diff
@@ -36,46 +38,67 @@ private UserCache(final Database database)
 		this.database = database;
 	}
 	
+	public User getUser (final UUID id)
+	{
+		return this.getUser("user_id", id);
+	}
+	
+	public User getUser (final String username)
+	{
+		return this.getUser("user_name", username);
+	}
 	
-	public User getUser(final String username)
+	private User getUser(final String columnName, final Object columnValue)
 	{
-		QueryRunner run = new QueryRunner(Database.getInstance().getDataSource());
-		List<User> users = null;
+		QueryRunner run = this.database.getQueryRunner();
+		List<User> foundUsers = null;
 		try
 		{
-			users = run.query(
-					"SELECT * FROM users WHERE user_name='?';",
-					new UserResultsSetHandler(), username);
+			foundUsers = run.query("SELECT * FROM users WHERE ?=?;",
+					new UserResultsSetHandler(), columnName, columnValue);
 		}
 		catch (SQLException e)
 		{
 			e.printStackTrace();
 		}
-		if (users == null || users.isEmpty())
+		if (foundUsers == null || foundUsers.isEmpty())
 		{
 			return null;
 		}
-		return users.get(0);
+		this.users.addAll(foundUsers);
+		return foundUsers.get(0);
+	}
+	
+	public void putUser(final User user) throws SQLException
+	{
+		AsyncQueryRunner run = this.database.getAsyncQueryRunner();
+		run.update("INSERT INTO users (user_id, user_name, user_pass, forename, surname, address_id) VALUES (?,?,?,?,?,?)"

```

Comment:
Could this use UPSERT? Also a comment explainging the possible results would be nice